server/src/main.rs

```diff
@@ -764,9 +857,31 @@ fn trace_stream(res: &mut Response<Streaming>, child: &mut child::Thread) -> io:
     res.write_all(b"\n]")?;
     Ok(terminated)
 }
-
+    
+// Stops the execution and terminates the process.  User has to start over
 /// POST /debug/:id/executions/:execution/stop -- Halts a running execution
-fn debug_execution_stop(req: Request, mut res: Response, _: Captures) {
-    *res.status_mut() = StatusCode::NotImplemented;
-    send(req, res, b"").unwrap();
+fn debug_execution_stop(caps: Captures, child: ChildThread) -> io::Result<Vec<u8>> {
+    let caps = caps.unwrap();
+    let _debug_id = caps[1].parse::<u64>()
+        .map_err(|e| io::Error::new(io::ErrorKind::NotConnected, e))?;
+    let execution_id = caps[2].parse::<i32>()
+        .map_err(|e| io::Error::new(io::ErrorKind::InvalidInput, e))?;
+
+    let child = child.lock().unwrap();
+    let child = child.as_ref()
+        .ok_or(io::Error::from(io::ErrorKind::NotConnected))?;
+
+    match child.execution {
+        Some((id, _)) if id == execution_id => {},
+        _ => return Err(io::Error::new(io::ErrorKind::NotFound, "no such execution")),
+    };
+
+    child.tx.send(ServerMessage::Quit).unwrap();
```

Comment:
Cancelling a process execution should not terminate the process- it should just pause it, so the user can place breakpoints or call functions. This should remove the differences between cancelling function/process executions.